src/openassetio-python/tests/package/test_version.py

```diff
@@ -0,0 +1,58 @@
+#
+# Copyright 2024 The Foundry Visionmongers Ltd
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+# http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""
+Tests that cover the version introspection methods
+"""
+
+# pylint: disable=invalid-name,redefined-outer-name
+# pylint: disable=too-few-public-methods
+# pylint: disable=missing-class-docstring,missing-function-docstring
+
+import re
+
+import openassetio
+
+# Update this value each release
+openassetio_version_string = "v1.0.0-beta.1.0"
+
+# Extract version numbers so we can test them
+pattern = r"v(\d+)\.(\d+)\.(\d+)-beta\.(\d+)\.(\d+)"
+match = re.search(pattern, openassetio_version_string)
```

Comment:
[consider][strong] Please move the regex logic to a fixture or function. Currently if it fails then it breaks at import time and will break test discovery before any tests even run.  Not the end of the world, but it's not super friendly.

To avoid recalculating with every test you can use e.g. `@pytest.fixture(scope="module")`.

Reply:
I didn't realise that. Good catch. https://github.com/OpenAssetIO/OpenAssetIO/pull/1247/commits/fc2e3e2fde2bcb7af53daf22b0a4b6086edba58b